app/lib/activitypub/activity/update.rb

```diff
@@ -32,4 +39,26 @@ def update_status
 
     ActivityPub::ProcessStatusUpdateService.new.call(@status, @json, @object, request_id: @options[:request_id])
   end
+
+  def allow_username_update?
+    updated_username_unique? && updated_username_confirmed?
+  end
+
+  def updated_username_unique?
+    account_proxy = @account.dup
+    account_proxy.username = @object['preferredUsername']
+    UniqueUsernameValidator.new.validate(account_proxy)
```

Comment:
This is exceedingly edge-casey, but just raising it because it occurred to me ... in theory the username value on the record could be valid at this check, then be changed, and be subsequently invalid when tried to save/update later on ... I guess that would blow up w/ validation error and be handled on whatever existing path does that.